src/nvim/regexp_nfa.c

```diff
@@ -6075,6 +6079,13 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
     if (got_int) {
```

Comment:
That's weird, both `fast_breakcheck` and `line_breakcheck` call `os_breakcheck`. Sounds like a race condition which we should address. Can you post the backtrace?